app/src/main/java/gholzrib/arctouchchallenge/ui/activities/MainActivity.java

```diff
@@ -0,0 +1,395 @@
+                    break;
+                case REQUEST_OPERATION_SEARCH:
+                    mLoadingDialog.setMessage(MESSAGE_LOADING_SEARCH_RESULTS);
+                    break;
+            }
+            if (!mLoadingDialog.isShowing() && nextPage == 1) {
+                mLoadingDialog.show();
+            }
+        }
+    }
+
+    @Override
+    public void onRequestEnds(int operation, boolean isSuccess, int code, String parsedData) {
+        if (isSuccess) {
+            try {
+                JSONObject jsonObject = new JSONObject(parsedData);
+                switch (code) {
+                    case Constants.RESPONSE_CODE_OK:
+                        handleSuccessfulRequestResults(operation, jsonObject);
+                        break;
+                    case Constants.RESPONSE_CODE_UNAUTHORIZED:
+                    case Constants.RESPONSE_CODE_NOT_FOUND:
+                        String statusMessage = null;
+                        int statusCode = 0;
+                        if (!jsonObject.isNull(Constants.RESPONSE_STATUS_MESSAGE))
+                            statusMessage = jsonObject.getString(Constants.RESPONSE_STATUS_MESSAGE);
+                        if (!jsonObject.isNull(Constants.RESPONSE_STATUS_CODE))
+                            statusCode = jsonObject.getInt(Constants.RESPONSE_STATUS_CODE);
+                        if (statusCode != 0) statusMessage = statusMessage + "(Code: " + statusCode + ")";
+                        if (statusMessage != null) Toast.makeText(this, statusMessage, Toast.LENGTH_SHORT).show();
+                        if (mLoadingDialog != null && mLoadingDialog.isShowing()) {
+                            mLoadingDialog.dismiss();
+                        }
+                        break;
+                }
+            } catch (JSONException e) {
+                Log.i(TAG, "JSONException: " + e.getMessage());
+            }
+        } else {
+            if (mLoadingDialog != null && mLoadingDialog.isShowing()) {
+                mLoadingDialog.dismiss();
+            }
+            Toast.makeText(this, Constants.ERROR_REQUEST_UNSUCCESSFUL, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    /**
+     * Changes the visibility of the No Internet Warning and RecyclerView according with the
+     * internet connection state
+     *
+     * @param hasConnection
+     */
+    private void setWarningAndListVisibilities(boolean hasConnection) {
+        if (hasConnection) {
+            mRvMovies.setVisibility(View.VISIBLE);
+            mLnrWarningNoInternet.setVisibility(View.GONE);
+        } else {
+            mRvMovies.setVisibility(View.GONE);
+            mLnrWarningNoInternet.setVisibility(View.VISIBLE);
+        }
+    }
+
+    /**
+     * Attempt to execute a request.
+     *
+     * @param additionalParams
+     * @param operation
+     */
+    private void attemptToExecuteRequest(String[] additionalParams, int operation) {
```

Comment:
There isn't a technical reason, it's just the way I usually organize the requests. A class to handle all requests sounds interesting, I will certainly think about that.